### PR TITLE
node 15.0.1

### PR DESCRIPTION
**Pull request by dio-az**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
-----

### CODE REVIEW

**Comment by diogoazevedos** [comment]
```
==> Downloading https://registry.npmjs.org/npm/-/npm-7.0.0.tgz
######################################################################## 100.0%
==> Downloading https://nodejs.org/dist/v15.0.0/node-v15.0.0.tar.gz
######################################################################## 100.0%
==> ./configure --prefix=/usr/local/Cellar/node/15.0.0 --without-npm --with-intl=system-icu
==> make install
==> node /private/tmp/node-20201020-66889-r6unch/node-v15.0.0/npm_bootstrap/bin/npm-cli.js install -ddd --glob
Last 15 lines from /Users/diogoazevedos/Library/Logs/Homebrew/node/03.node:
npm verb argv "/usr/local/Cellar/node/15.0.0/bin/node" "/private/tmp/node-20201020-66889-r6unch/node-v15.0.0/npm_bootstrap/bin/npm-cli.js" "install" "-ddd" "--global" "--prefix=/usr/local/Cellar/node/15.0.0/libexec" "/Users/diogoazevedos/Library/Caches/Homebrew/downloads/795a7354ab674b9b41e3574251623fea3c9a4aa987a13c81b5224e73f1876f2e--npm-7.0.0.tgz"
npm verb node v15.0.0
npm verb npm  v7.0.0
npm ERR! code ENOENT
npm ERR! syscall lstat
npm ERR! path /usr/local/Cellar/node/15.0.0/libexec
npm ERR! errno -2
npm ERR! enoent ENOENT: no such file or directory, lstat '/usr/local/Cellar/node/15.0.0/libexec'
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent 
npm verb exit -2
npm timing npm Completed in 2767ms

npm ERR! A complete log of this run can be found in:
npm ERR!     /private/tmp/node-20201020-66889-r6unch/node-v15.0.0/.brew_home/.npm/_logs/2020-10-20T16_34_36_133Z-debug.log
```

**Comment by SMillerDev** [comment]
Please don't add a node@14 formula unless it fulfills all requirements in https://docs.brew.sh/Versions

File: Formula/node.rb [hunk 1]
```diff
@@ -49,6 +49,8 @@ def install
 
     bootstrap = buildpath/"npm_bootstrap"
     bootstrap.install resource("npm")
+    # These dirs must exists before npm install.
+    mkdir_p libexec/"lib"
```
Comment (diogoazevedos) [inline Formula/node.rb:53]: Should I add this to https://github.com/Homebrew/brew/blob/24ae318a3d5f0519f5fe29d08e01e8a044d99d87/Library/Homebrew/language/node.rb#L39-L60, as the same issue happened for `bcoin`?

**Comment by billinghamj** [comment]
This might also be a good time to remove the OpenSSL ASM patch:

https://github.com/Homebrew/homebrew-core/pull/63170/files#diff-074ce911bbf848cc1258962b3527853528213ced30b268e5de23309d856b158dR39-R41

I don't think that should be required any more - though needs to be validated

**Review by SMillerDev — DISMISSED** [review]
Thanks @diogoazevedos  ! Without contributions like yours it'd be impossible to keep homebrew going with the high standards that users have come to expect from the project. You can feel good knowing that you've made the world a tiny bit better for homebrew users around the world! 👍 🎉

**Comment by janosh** [comment]
Thanks @diogoazevedos! Looking forward to this.

**Review by chenrui333 — APPROVED** [review]

**Pull request closed without merging**